tests/fixtures/given.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fixtures
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/numaproj-labs/numaplane/pkg/apis/numaplane/v1alpha1"
+	planepkg "github.com/numaproj-labs/numaplane/pkg/client/clientset/versioned/typed/numaplane/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/yaml"
+)
+
+type Given struct {
+	t             *testing.T
+	restConfig    *rest.Config
+	kubeClient    kubernetes.Interface
+	gitSyncClient planepkg.GitSyncInterface
+	gitSync       *v1alpha1.GitSync
+}
+
+// create GitSync using raw YAML or @filename
+func (g *Given) GitSync(text string) *Given {
+	g.t.Helper()
+	g.gitSync = &v1alpha1.GitSync{}
+	g.readResource(text, g.gitSync)
+	g.addLabel()
+	return g
+}
+
+func (g *Given) WithGitSync(gs *v1alpha1.GitSync) *Given {
+	g.t.Helper()
+	g.gitSync = gs
+	g.addLabel()
+	return g
+}
+
+func (g *Given) addLabel() {
```

Comment:
nit: let's rename to `addE2ELabel()` since that's what it is (right now sounds like a generic label)